Grantors - POST and DELETE

AkshayaKomaragiri opened the pull request.

TusharW4ni commented:
- Can you please change the API route to be /grantors/delete/[id]/route.ts instead of /grantors/[id]/delete/route.ts
- I understand that you have created the `prismaSoftDelete` function to update the status on grantors to false. Is there a reason why you are not directly doing it in the API?

AkshayaKomaragiri commented:
- I changed the API route to delete/[id]. 
- I didn't put it in the API is because the prisma extension that I added can be used for anything, not just grantors. So, in the future if we need something like this again we just use the extension and it would decrease redundancy.

TusharW4ni commented:
I'm not really sure the extension will be too helpful, but if you feel it will be, we can use it no problem.
